bug: lookup 0.0.0: no such host

### Describe the bug
`[GIN-debug] [ERROR] listen tcp: lookup 0.0.0: no such host`

### Expected behavior
Gin to start on 0.0.0.0 with port 8080

### How to reproduce?
Run code in docker

```yaml
services:
  go:
    build:
      context: .
      dockerfile_inline: |
        FROM golang:1.22-alpine
        RUN apk add git
        WORKDIR /app
        RUN git clone https://github.com/logto-io/go.git
        WORKDIR /app/go/gin-sample
        RUN go mod download
        CMD go run .
    ports:
      - 8080:8080
```

--- a/.gitignore
+++ b/.gitignore
@@ -1,1 +1,2 @@
 /build
+.env

--- a/gin-sample/.env.example
+++ b/gin-sample/.env.example
@@ -1,0 +1,6 @@
+ENDPOINT=http://localhost:3001
+APP_ID=
+APP_SECRET=
+RESOURCES=https://default.logto.app/api
+REDIRECT_URI=http://localhost:8080/sign-in-callback
+POST_SIGN_OUT_REDIRECT_URI=http://localhost:8080

--- a/gin-sample/compose.yml
+++ b/gin-sample/compose.yml
@@ -9,14 +9,9 @@ services:
         RUN git clone https://github.com/logto-io/go.git
         WORKDIR /app/go/gin-sample
         RUN go mod download
-
-        ENV ENDPOINT=<your-logto-endpoint>
-        ENV APP_ID=<your-app-id>
-        ENV APP_SECRET=<your-app-secret>
-        ENV RESOURCES=<your-resources>
-        ENV CALLBACK_URI=<your-callback-uri>
-        ENV POST_LOGOUT_REDIRECT_URI=<your-post-logout-redirect-uri>
-
-        CMD go run .
+        RUN CGO_ENABLED=0 GOOS=linux go build -o /logto-go
+    command: /logto-go
+    env_file:
+      - .env
     ports:
       - 8080:8080

--- a/gin-sample/compose.yml
+++ b/gin-sample/compose.yml
@@ -1,0 +1,22 @@
+services:
+  logto-go:
+    build:
+      context: .
+      dockerfile_inline: |
+        FROM golang:1.22-alpine
+        RUN apk add git
+        WORKDIR /app
+        RUN git clone https://github.com/logto-io/go.git
+        WORKDIR /app/go/gin-sample
+        RUN go mod download
+
+        ENV ENDPOINT=<your-logto-endpoint>
+        ENV APP_ID=<your-app-id>
+        ENV APP_SECRET=<your-app-secret>
+        ENV RESOURCES=<your-resources>
+        ENV CALLBACK_URI=<your-callback-uri>
+        ENV POST_LOGOUT_REDIRECT_URI=<your-post-logout-redirect-uri>
+
+        CMD go run .
+    ports:
+      - 8080:8080

--- a/gin-sample/main.go
+++ b/gin-sample/main.go
@@ -2,6 +2,7 @@ package main
 
 import (
 	"net/http"
+	"os"
 
 	"github.com/gin-contrib/sessions"
 	"github.com/gin-contrib/sessions/memstore"
@@ -15,10 +16,10 @@ var (
 
 func main() {
 	logtoConfig := &client.LogtoConfig{
-		Endpoint:  "<your-logto-endpoint>", // E.g. "http://localhost:3001"
-		AppId:     "<your-app-id>",
-		AppSecret: "<your-app-secret>",
-		Resources: []string{"your-resource"}, // E.g. "https://default.logto.app/api"
+		Endpoint:  os.Getenv("ENDPOINT"),
+		AppId:     os.Getenv("APP_ID"),
+		AppSecret: os.Getenv("APP_SECRET"),
+		Resources: []string{os.Getenv("RESOURCES")},
 	}
 
 	store := memstore.NewStore([]byte("secret"))
@@ -51,7 +52,7 @@ func main() {
 	router.GET("/sign-in", func(ctx *gin.Context) {
 		session := sessions.Default(ctx)
 		logtoClient := client.NewLogtoClient(logtoConfig, &SessionStorage{session: session})
-		signInUri, err := logtoClient.SignIn("http://localhost:8080/sign-in-callback")
+		signInUri, err := logtoClient.SignIn(os.Getenv("CALLBACK_URI"))
 		if err != nil {
 			ctx.String(http.StatusInternalServerError, err.Error())
 			return
@@ -89,7 +90,7 @@ func main() {
 		session := sessions.Default(ctx)
 		logtoClient := client.NewLogtoClient(logtoConfig, &SessionStorage{session: session})
 
-		signOutUri, signOutErr := logtoClient.SignOut("http://localhost:8080")
+		signOutUri, signOutErr := logtoClient.SignOut(os.Getenv("POST_LOGOUT_REDIRECT_URI"))
 
 		if signOutErr != nil {
 			ctx.String(http.StatusOK, signOutErr.Error())

--- a/gin-sample/main.go
+++ b/gin-sample/main.go
@@ -3,6 +3,7 @@ package main
 import (
 	"net/http"
 	"os"
+	"strings"
 
 	"github.com/gin-contrib/sessions"
 	"github.com/gin-contrib/sessions/memstore"
@@ -16,10 +17,15 @@ var (
 
 func main() {
 	logtoConfig := &client.LogtoConfig{
+		// see .env.example for more details and examples
 		Endpoint:  os.Getenv("ENDPOINT"),
 		AppId:     os.Getenv("APP_ID"),
 		AppSecret: os.Getenv("APP_SECRET"),
-		Resources: []string{os.Getenv("RESOURCES")},
+		Resources: []string{},
+	}
+	resources, present := os.LookupEnv("RESOURCES")
+	if present {
+		logtoConfig.Resources = strings.Split(resources, ",")
 	}
 
 	store := memstore.NewStore([]byte("secret"))
@@ -52,7 +58,7 @@ func main() {
 	router.GET("/sign-in", func(ctx *gin.Context) {
 		session := sessions.Default(ctx)
 		logtoClient := client.NewLogtoClient(logtoConfig, &SessionStorage{session: session})
-		signInUri, err := logtoClient.SignIn(os.Getenv("CALLBACK_URI"))
+		signInUri, err := logtoClient.SignIn(os.Getenv("REDIRECT_URI"))
 		if err != nil {
 			ctx.String(http.StatusInternalServerError, err.Error())
 			return
@@ -90,7 +96,7 @@ func main() {
 		session := sessions.Default(ctx)
 		logtoClient := client.NewLogtoClient(logtoConfig, &SessionStorage{session: session})
 
-		signOutUri, signOutErr := logtoClient.SignOut(os.Getenv("POST_LOGOUT_REDIRECT_URI"))
+		signOutUri, signOutErr := logtoClient.SignOut(os.Getenv("POST_SIGN_OUT_REDIRECT_URI"))
 
 		if signOutErr != nil {
 			ctx.String(http.StatusOK, signOutErr.Error())

--- a/gin-sample/main.go
+++ b/gin-sample/main.go
@@ -144,5 +144,5 @@ func main() {
 		ctx.Data(http.StatusOK, ContentTypeHtml, []byte(unauthorizedPage))
 	})
 
-	router.Run("0.0.0:8080")
+	router.Run("0.0.0.0:8080")
 }

